contentcuration/contentcuration/static/js/administration/router.js

```diff
@@ -0,0 +1,293 @@
+var Backbone = require("backbone");
+var Models = require("edit_channel/models");
+var _ = require("underscore");
+
+// this is a function so that the defaults don't get overwritten
+const GET_DEFAULT_ROUTES = () => ({
+	'users': {name: 'users', page: 1, sortKey: "email", sortOrder: "ascending"},
+	'channels': {name: 'channels', page: 1, sortKey: "name", sortOrder: "ascending"},
+})
+
+const CHANNEL_FILTERS = {
+	all: {
+		label: "All",
+		queryParams: {}
+	}, 
+	live: {
+		label: "Live",
+		queryParams: { live: "True"},
+	},
+	published: {
+		label: "Published",
+		queryParams: { published: "True"},
+	},
+	public: {
+		label: "Public",
+		queryParams: { public: "True"},
+	},
+	private: {
+		label: "Private",
+		queryParams: { public: "False" },
+	},
+	can_edit: {
+		label: "My Channels",
+		queryParams: { editable: "True" },
+	},
+	staged: {
+		label: "Needs Review",
+		queryParams: { staged: "True" },
+	},
+	sushichef: {
+		label: "Sushi Chef",
+		queryParams: { ricecooker_version__isnull: "False" },
+	},
+	deleted: {
+		label: "Deleted",
+		queryParams: { deleted: "True" },
+	},
+}
+
+const CHANNEL_SORT_FILTERS = {
+	name: {
+		label: "Name",
+		selected: true,
+	},
+	id: {
+		label: "Channel ID",
+	},
+	priority: {
+		label: "Priority",
+	},
+	users: {
+		label: "# of Users",
+	},
+	items: {
+		label: "# of Items",
+	},
+	created: {
+		label: "Date Created",
+	}
+}
+
+const USER_FILTERS = {
+	all: {
+		label: "All",
+		queryParams: {},
+		selected: true,
+	},
+	activated: {
+		label: "Active",
+		queryParams: { is_active: "True" },
+	},
+	not_activated: {
+		label: "Inactive",
+		queryParams: { is_active: "False" },
+	}, 
+	admins: {
+		label: "Admins",
+		queryParams: { is_admin: "True" },
+	},
+	is_chef: {
+		label: "Sushi Chefs",
+		queryParams: { ricecooker_version: ">0.0.0" },
+	}
+}
+
+const USER_SORT_FILTERS = {
+	email: {
+		label: "Email",
+	},
+	first_name: {
+		label: "First Name",
+	},
+	last_name: {
+		label: "Last Name",
+	},
+	date_joined: {
+		label: "Date Joined",
+	},
+	channels: {
+		label: "# of Channels",
+	},
+}
+
+const SORT_ORDER_OPTIONS = {
+	ascending: {
+		label: "Ascending",
+		selected: true,
+	},
+	descending: {
+		label: "Descending",
+	},
+}
+
+var AdministrationRouter = Backbone.Router.extend({
+
+    routes: {
+		"": "default",
+		"/": "default",
+		"users/": "users",
+		"channels/": "channels",
+        "users(/filter/:filter)(/sort/:key-:order)(/search/:search)(/p:page)(/:pagesize-per-page)":        "users",    // #users
+        "channels(/filter/:filter)(/sort/:key-:order)(/search/:search)(/p:page)(/:pagesize-per-page)":     "channels",    // #channels
+    },
+	execute(callback, args, name){
```

Comment:
This function pre-processes each route to see if we need to switch tabs or re-render.